Issue/16 material design

tonyr59h opened the pull request:
Updating keyboard and preferences to follow Material design guidelines more closely.

Layouts for larger screens limits the size of the keypad, it was hideously huge before.
Ripple effect added to keypad buttons.
Removed the WordPress logo in favor of something more generic.

After the merge we'll need to bump the version and do a new release so the changes can be pulled into the WordPress-Android project.

cc @roundhill

tonyr59h committed:
Updating build tools and compiled SDK version.

tonyr59h committed:
Removing WordPress logo in favor of more generic icon.

tonyr59h committed:
Disabling rotation for smaller devices due to layout constraints.

Took a hint from the Google Wallet app. With rotation enabled in phones the user would have to scroll to reach all the buttons for input.

tonyr59h committed:
Updating string and color resources.

tonyr59h committed:
Typo in layout.

tonyr59h committed:
Using an EditText in place of individual passcode fields.

tonyr59h committed:
Removing landscape layout for now and adding many small layout changes.

tonyr59h committed:
Using XML drawable for logo

tonyr59h committed:
Adding ripple effect on keypad button presses.

tonyr59h committed:
Fixing pin validation.

tonyr59h committed:
Adding layout file for tablets that limits the width of the keyboard.

roundhill commented:
What if we used this icon from the material icon set in place of the 'Del' key? Maybe it'd be harder to theme, though...

![ic_backspace_black_24dp](https://cloud.githubusercontent.com/assets/789137/7755768/7f60b808-ffab-11e4-9c3f-28fbbadb26f4.png)

roundhill commented on library/res/layout-sw600dp/app_passcode_keyboard.xml:
```diff
@@ -1,78 +1,62 @@
-            </EditText>
-        </LinearLayout>
+            android:layout_marginRight="38dp"
+            android:hint="@string/passcodelock_hint"
```
Do you think we even need the hint? It says 'Enter PIN code' already right above it.

roundhill replied:
Or we could kill the label that's above it and just use 'Enter PIN code' as the hint?

tonyr59h commented:
> What if we used this icon from the material icon set in place of the 'Del' key?

Much preferred over the existing 'Del' text.

tonyr59h committed:
Removing PIN field hint, can override string resource to add one.

tonyr59h committed:
Using an icon to represent Del and increasing font size on keypad text.

Define @dimen/passcodelock_button_text_size to override the keypad text size.

The icon is tinted to match the text color.

daniloercoli commented:
Could you please update the instructions to change the logo:

<blockquote>Customization
If you want to customize the pinlock icon, the one available in the unlock screen, add a file named passcode_logo.png in res/drawable.
</blockquote>

Also, the default logo doesn't look so good on hi-res devices
![screen shot 2015-05-25 at 11 43 50](https://cloud.githubusercontent.com/assets/518232/7795251/5d4aa01e-02d3-11e5-9a05-156905907732.png)

daniloercoli commented:
The "Ripple effects" for buttons on the 1st row doesn't have the correct z-index:

![passcode-ripple-issue-01](https://cloud.githubusercontent.com/assets/518232/7795372/a4a98d48-02d4-11e5-86b4-d295e459fdc8.gif)

daniloercoli commented:
One more things on buttons, The ripple effects is not available in the library. I've to apply the WordPress Calypso style to get it working. Maybe worth to style the buttons in the library too.

tonyr59h committed:
Confining ripple animation to view boundary and updating README.

tonyr59h committed:
Adding more logo assets for different screen sizes.

tonyr59h committed:
Updating style so that ripple animation is visible.

tonyr59h committed:
Adding missing compatibility library dependency; used for styling.

tonyr59h commented:
> The "Ripple effects" for buttons on the 1st row doesn't have the correct z-index

After looking at the rest of the WordPress-Android project I've decided to confine the ripple animation within the view. You can see this effect in the MasterBar and Settings pages (and more, I'm sure). With that change in place the z-order is no longer an issue.

> I've to apply the WordPress Calypso style to get it working.

I think this is because we have not defined the colors used by the animation. I've added the required colors and the compatibility library.

daniloercoli commented:
Not sure if it's an issue of the library or of the app, but the toolbar is missing of the back/home button in `PasscodePreferencesActivity`
![screen shot 2015-05-27 at 11 02 47](https://cloud.githubusercontent.com/assets/518232/7832565/58401f34-0460-11e5-96ec-2f7e294e7c0a.png)

daniloercoli commented:
I've investigated a bit about the missing Toolbar in PasscodePreferencesActivity and seems there isn't a n easy solution for that. The simplest I've found is this one: http://mrengineer13.com/how-to-add-an-actionbar-to-your-settingsactivity-with-appcompat-v21/ or rewriting PasscodePreferencesActivity with ActionBarActivity  + PreferenceFragment. 

/cc @maxme

maxme commented:
> PasscodePreferencesActivity with ActionBarActivity + PreferenceFragment.

Probably the best solution. That's what we did in wpandroid for the same reason.

daniloercoli commented:
I just found out that the toolbar appears in the app if I removed the 
`android:theme="@style/CalypsoTheme"` to the PasscodePreferencesActivity. 
Maybe it's not a problem of the library, but something we need to tackle in the app.

![screen shot 2015-05-27 at 15 26 46](https://cloud.githubusercontent.com/assets/518232/7837006/cd1bfeda-0484-11e5-87ac-0624fa674fbb.png)

tonyr59h commented:
I've styled it in the WordPress-Android project to better match the other preference screens.

> PasscodePreferencesActivity with ActionBarActivity + PreferenceFragment

If we want to refine it more and implement it like this I'm not opposed. But in my opinion the style in WPAndroid is very close now.

daniloercoli commented:
Styling the Preference ActionBar in wp-android seemed to work fine at first glance, but I found issues with the Ripple effect, the label, and the icon. 
Icon - It's visible in the bar. 
Label - It's clickable in PinLock Screen, while it's not in the app.

![passcode-ripple-issue-02](https://cloud.githubusercontent.com/assets/518232/7857653/6cbdfbfc-0532-11e5-8da6-a8a5ba45d67b.gif)

Also, introducing new styles that are just used in one PinLock screen seems a bit overkill to me. If we need to change the Action bar in the app, we then need to update these new styles...(i'm not super expert with styles and UI stuff, maybe this is the only way to achieve the goal).

My impressions here is that we need to switch to 

> PasscodePreferencesActivity with ActionBarActivity + PreferenceFragment

as @maxme suggested.

I'd to hear what @nbradbury has to say about this.

nbradbury commented:
> I'd [like] to hear what @nbradbury has to say about this.

Trying to match the style will likely be an endless battle. Would it be possible to move "Turn PIN lock on" and "Change PIN" to our main preferences screen? That way the only additional screen would be to change the PIN, which is fullscreen and doesn't need to match.

tonyr59h commented:
@nbradbury thanks for the input, seems we've been looking at it at the wrong angle. I agree, it's usually a good idea to remove a level of indirection with the UI. I'll try moving those options up a level unless there's any objections.

daniloercoli commented:
> I'll try moving those options up a level unless there's any objections.

:+1:

tonyr59h committed:
Switching from PreferenceActivity to PreferenceFragment implementation.

* Upped build tools to latest version
* Set min SDK to 14 and compiling against latest (22)
* Cleaning up some code

maxme committed:
add sonatype configuration

maxme committed:
update README

maxme commented:
I added sonatype configuration to that repo in this branch, so we can publish the artifact on maven central.

daniloercoli committed:
Update readme.md

daniloercoli commented:
I've updated the file readme.md to reflect the switch from PreferenceActivity to PreferenceFragment.

@maxme Can you take a look it? If everything is OK for you,  we're ready to merge this PR, and publish the new lib version 0.0.7 to Maven.

tonyr59h commented:
Thanks for updating the docs guys! I'll go ahead with the merge and push a new release to Maven Central so we can do a 4.2 code freeze.
